internal/app/grpcclient/impl.go

```diff
@@ -275,30 +284,32 @@ func (i *invoker) bidiStream(
 		}
 	}
 
-	// If we received an error in any of the send logic or full-duplex reads, then exit
-	if protoErr != nil {
-		result.Error = protoErr
-		return result, nil
-	}
+	var hdr metadata.MD
+	defer func() {
+		if result != nil {
+			// Set headers and trailers from the stream
+			result.ResponseHeaders = grpcutil.ConvertMetadataToProtoHeader(hdr)
+			// Trailers are
+			result.ResponseTrailers = grpcutil.ConvertMetadataToProtoHeader(stream.Trailer())
+		}
+	}()
 
 	// Sends are done, close the send side of the stream
 	if err := stream.CloseSend(); err != nil {
-		result.Error = grpcutil.ConvertGrpcToProtoError(err)
-		return result, nil
+		return nil, err
 	}
 
 	// Once the send side is closed, header metadata is ready to be read
-	hdr, err := stream.Header()
+	hdr, err = stream.Header()
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		if result != nil {
-			// Set headers and trailers from the stream
-			result.ResponseHeaders = grpcutil.ConvertMetadataToProtoHeader(hdr)
-			result.ResponseTrailers = grpcutil.ConvertMetadataToProtoHeader(stream.Trailer())
-		}
-	}()
```

Comment:
This had to be rearranged a bit because in instances where we returned early (i.e. line 281 above), the `defer` wasn't defined yet, so headers and trailers were not getting set.